node/cn/api.go

```diff
@@ -124,87 +67,12 @@ func NewPrivateMinerAPI(e *CN) *PrivateMinerAPI {
 	return &PrivateMinerAPI{e: e}
 }
 
-// Start the miner with the given number of threads. If threads is nil the number
-// of workers started is equal to the number of logical CPUs that are usable by
-// this process. If mining is already running, this method adjust the number of
-// threads allowed to use.
-func (api *PrivateMinerAPI) Start(threads *int) error {
-	// Set the number of threads if the seal engine supports it
-	if threads == nil {
-		threads = new(int)
-	} else if *threads == 0 {
-		*threads = -1 // Disable the work from within
-	}
-	type threaded interface {
-		SetThreads(threads int)
-	}
-	if th, ok := api.e.engine.(threaded); ok {
-		logger.Info("Updated mining threads", "threads", *threads)
-		th.SetThreads(*threads)
-	}
-	// Start the miner and return
-	if !api.e.IsMining() {
-		// Propagate the initial price point to the transaction pool
-		api.e.lock.RLock()
-		price := api.e.gasPrice
-		api.e.lock.RUnlock()
-
-		if price.Cmp(api.e.txPool.GasPrice()) == 0 {
-			return api.e.StartMining(true)
-		} else {
-			logger.Error("PrivateMinerAPI Start: Invalid unit price from API", "TxPool UnitPrice", api.e.txPool.GasPrice(), "API UnitPrice", price)
-			return blockchain.ErrInvalidUnitPrice
-		}
-	}
-	return nil
-}
-
-// Stop the miner
-func (api *PrivateMinerAPI) Stop() bool {
-	type threaded interface {
-		SetThreads(threads int)
-	}
-	if th, ok := api.e.engine.(threaded); ok {
-		th.SetThreads(-1)
-	}
-	api.e.StopMining()
-	return true
-}
-
-// SetExtra sets the extra data string that is included when this miner mines a block.
-func (api *PrivateMinerAPI) SetExtra(extra string) (bool, error) {
-	if err := api.e.Miner().SetExtra([]byte(extra)); err != nil {
-		return false, err
-	}
-	return true, nil
-}
-
-// SetGasPrice sets the minimum accepted gas price for the miner.
-func (api *PrivateMinerAPI) SetGasPrice(gasPrice hexutil.Big) bool {
-	if api.e.txPool.GasPrice().Cmp((*big.Int)(&gasPrice)) != 0 {
-		logger.Debug("PrivateMinerAPI.SetGasPrice", "TxPool UnitPrice", api.e.txPool.GasPrice(), "Given UnitPrice", gasPrice.ToInt())
-		return false
-	}
-
-	api.e.lock.Lock()
-	api.e.gasPrice = (*big.Int)(&gasPrice)
-	api.e.lock.Unlock()
-
-	api.e.txPool.SetGasPrice((*big.Int)(&gasPrice))
-	return true
-}
-
 // SetRewardbase sets the rewardbase of the CN.
 func (api *PrivateMinerAPI) SetRewardbase(rewardbase common.Address) bool {
```

Comment:
Is this the only function remained in PrivateMinterAPI?
If it is, I think we can remove it too. 
@andybclee What do you think?

Reply:
@andybclee Could you please check the above comment?

Reply:
@andybclee Could you please check the comment?

Reply:
@kjhman21 @ehnuje The `RewardBase` is only meaningful before the addressbook contract activated. After the activation, the address information is retrieved from the contract, so setting a rewardbase is meaningless at the moment. 

In whatever reason, if we need the rewardbase again, it can still be set as a parameter when starting up `kcn`

So I think it is okay to be removed.